gerrit/server.go

```diff
@@ -216,3 +222,178 @@ func (s *Server) PostCheck(changeID string, psID int, input *CheckInput) (*Check
+	q.Add("q", "is:open")
+	u.RawQuery = q.Encode()
+
+	content, err := s.Get(&u)
+	if err != nil {
+		return err
+	}
+
+	var out []*PendingSubmitInfo
+	if err := Unmarshal(content, &out); err != nil {
+		return err
+	}
+
+	var patchsets []*PendingSubmitInfo
+
+	for i := 0; i < len(out); i++ {
+		// Ignore merge conflicts, patchsets without required labels, and patchsets currently being handled by Jarvis
+		if out[i].Mergeable == true && out[i].Subittable == true && !contains(out[i].Hashtags, JarvisMergeHashtag) {
+			patchsets = append(patchsets, out[i])
+		}
+	}
+
+	log.Printf("Sending %d patch sets to merge", len(patchsets))
+
+	for _, obj := range patchsets {
+
+		// Find the patch set number by checking the number of revisions that have been make to the change
+		obj.PatchsetNumber = len(obj.Revisions)
+
+		if err = s.PostHashtag(obj); err != nil {
+			log.Printf("PostHashtag Error: %v", err)
+			return err
+		}
+
+		if err = s.CallPipeline(obj); err != nil {
+			log.Printf("CallPipeline Error: %v", err)
+			return err
+		}
+	}
+
+	return nil
+}
+
+type PendingSubmitInfo struct {
+	Id              string                 `json:"id"`
+	Project         string                 `json:"project"`
+	Branch          string                 `json:"branch"`
+	Hashtags        []string               `json:"hashtags"`
+	ChangeId        string                 `json:"change_id"`
+	ChangeNumber    int                    `json:"_number"`
+	Subject         string                 `json:"subject"`
+	Status          string                 `json:"status"`
+	Created         Timestamp              `json:"created"`
+	Updated         Timestamp              `json:"updated"`
+	SubmitType      string                 `json:"submit_type"`
+	Mergeable       bool                   `json:"mergeable"`
+	Subittable      bool                   `json:"submittable"`
+	CurrentRevision string                 `json:"current_revision"`
+	Revisions       map[string]interface{} `json:"revisions"`
+	PatchsetNumber  int
+}
+
+type HashtagPayload struct {
+	Add    []string `json:"add"`
+	Remove []string `json:"remove"`
+}
+
+type TektonMergePayload struct {
+	RepoRoot       string `json:"repoRoot"`
+	Project        string `json:"project"`
+	ChangeNumber   string `json:"changeNumber"`
+	PatchSetNumber string `json:"patchSetNumber"`
+	CheckerUUID	   string `json:"checkerUUID"`
+}
+
+func (s *Server) PostHashtag(patchset *PendingSubmitInfo) error {
+	u := s.URL
+	u.Path = path.Join(u.Path, fmt.Sprintf("a/changes/%s/hashtags/", patchset.ChangeId)) + "/"
+	hashtagPayload := HashtagPayload{
+		Add:    []string{JarvisMergeHashtag},
+		Remove: []string{},
+	}
+	body, err := json.Marshal(hashtagPayload)
+	if err != nil {
+		return err
+	}
+	_, err = s.PostPath(u.Path, "application/json", body)
+	if err != nil {
+		// return err
+		log.Printf("Error: %v", err)
+	}
+
+	return nil
+}
+
+func (s *Server) CallPipeline(patchset *PendingSubmitInfo) error {
+	// TODO pull this value from flag
```

Comment:
please fix this prior to merge - hardcoded URLs etc shouldn't end up in the connector.